tools/cell_census_builder/validate.py

```diff
@@ -74,58 +78,85 @@ def validate_all_soma_objects_exist(soma_path: str, experiment_builders: List[Ex
         assert datetime.fromisoformat(census.metadata["created_on"])
         assert "git_commit_sha" in census.metadata
 
-        for name in [CENSUS_INFO_NAME, CENSUS_DATA_NAME]:
-            assert soma.Collection.exists(census[name].uri)
-            assert census[name].soma_type == "SOMACollection"
+        with tiledb.Group(census.uri) as census_group:
+            for name in [CENSUS_INFO_NAME, CENSUS_DATA_NAME]:
+                assert soma.Collection.exists(census[name].uri)
+                assert census[name].soma_type == "SOMACollection"
+                assert census_group.is_relative(name), error_tiledb_group
 
         census_info = census[CENSUS_INFO_NAME]
-        for name in [CENSUS_DATASETS_NAME, CENSUS_SUMMARY_NAME, CENSUS_SUMMARY_CELL_COUNTS_NAME]:
-            assert name in census_info, f"`{name}` missing from census_info"
-            assert soma.DataFrame.exists(census_info[name].uri)
-            assert census_info[name].soma_type == "SOMADataFrame"
+        with tiledb.Group(census_info.uri) as census_info_group:
+            for name in [CENSUS_DATASETS_NAME, CENSUS_SUMMARY_NAME, CENSUS_SUMMARY_CELL_COUNTS_NAME]:
+                assert name in census_info, f"`{name}` missing from census_info"
+                assert soma.DataFrame.exists(census_info[name].uri)
+                assert census_info[name].soma_type == "SOMADataFrame"
+                assert census_info_group.is_relative(name), error_tiledb_group
 
         assert sorted(census_info[CENSUS_DATASETS_NAME].keys()) == sorted(CENSUS_DATASETS_COLUMNS + ["soma_joinid"])
         assert sorted(census_info[CENSUS_SUMMARY_CELL_COUNTS_NAME].keys()) == sorted(
             list(CENSUS_SUMMARY_CELL_COUNTS_COLUMNS) + ["soma_joinid"]
         )
         assert sorted(census_info[CENSUS_SUMMARY_NAME].keys()) == sorted(["label", "value", "soma_joinid"])
 
+        # verify required dataset fields are set
+        census_info[CENSUS_DATASETS_NAME]
+        df: pd.DataFrame = census_info[CENSUS_DATASETS_NAME].read().concat().to_pandas()
+        assert not all(df["collection_id"].isin([""]))
```

Comment:
I personally found this test a bit confusing. Totally optional, but do you think it would be clearer as:

```python
    (df["collection_id"] != "").all()
```

Reply:
alternatively, you could do all four tests together as:
```python
(df[["collection_id","collection_doi","collection_name","dataset_title"]] != "").all().all()
```

Reply:
I think the first suggestion is cleaner.